resources/rules/pre_identifier_merge/step_b_id_exact_match/datasource_xref/eg_gene_info_rules.clj

```diff
@@ -1,10 +1,10 @@
-;; EG gene id to other gene id mappings
+;; --------------------------------
+;; --------- EG Gene Info ---------
+;; --------------------------------
 `{:name "eg-gene-exact-mapping-assertion"
- :head ((?/geneid skos/exactMatch ?/otherice))
-  :body
-  (~@(kabob/rtv _/record
-      iaoeg/EntrezGeneInfoFileData_geneIDDataField1 ?/geneid
-      iaoeg/EntrezGeneInfoFileData_dbXrefsDataField1 ?/otherice)
-   (?/otherice kiao/denotesSubClassOf obo/SO_0000352))} ;dna
-    ;;rdf/type kiao/DNAIdentifier))}
-
+  :description "This rule asserts an exact match between eg genes and other gene ids"
+  :head ((?/geneid skos/exactMatch ?/otherice))
+  :body ((?/record iaoeg/EntrezGeneInfoFileData_geneIDDataField1 ?/geneid)
+         (?/record iaoeg/EntrezGeneInfoFileData_dbXrefsDataField1 ?/otherice)
+         (?/otherice kiao/denotesSubClassOf obo/SO_0000352)) ; SO:dna
+  }
```

Comment:
I really like the removal of the ~@(kabob/rtv macros. I think rtv stood for 'record-template-value'. Under the hood it got expanded into more than the two triples shown here. For example, I think the following:
```
(~@(kabob/rtv _/record
       iaoeg/EntrezGeneInfoFileData_geneIDDataField1 ?/geneid
       iaoeg/EntrezGeneInfoFileData_dbXrefsDataField1 ?/otherice)
```

was actually expanded to:

```
(?/record obo/BFO_0000051 ?/fv1)
(?/fv1 kiao/hasTemplate iaoeg/EntrezGeneInfoFileData_geneIDDataField1)
(?/fv1 obo/IAO_0000219 ?/geneid)
(?/record obo/BFO_0000051 ?/fv2)
(?/fv2 kiao/hasTemplate iaoeg/EntrezGeneInfoFileData_dbXrefsDataField1)
(?/fv2 obo/IAO_0000219 ?/otherice)
```

So, the updated version of this rule (I think) should look like:

```
:head((?/gene_identifier skos/exactMath ?/other_identifier))
:body((?/record rdf/type ccp/IAO_EXT_0000681) ;; ccp:NCBI gene info record
      (?/record obo/BFO_0000051 ?/gene_id_field_value)
      (?/gene_id_field_value rdf/type ccp/IAO_EXT_0000876) ;; ccp:NCBI gene info record - gene identifier field value
      (?/gene_id_field_value rdf/type ?/gene_identifier)
      (?/gene_identifier [rdfs/subClassOf *] ccp/IAO_EXT_0000182) ;; ccp:DNA identifier
      (?/record obo/BFO_0000051 ?/db_xref_field_value) 
      (?/db_xref_field_value rdf/type ccp/IAO_EXT_0000880) ;; ccp:NCBI gene info record - database cross-reference field value
      (?/db_xref_field_value rdf/type ?/other_identifier)
      (?/other_identifier [rdfs/subClassOf *] ccp/IAO_EXT_0000182)) ;; ccp:DNA identifier
```

The final product expands the rtv macro, replaces the old field value names with IAO_EXT concepts, and replaces the use of kiao/denotesSubClassOf with the `rdfs:subClassOf* IAO_EXT_0000182` constraints. The last constraint should act as a sanity check to help us avoid collapsing identifiers that are not of the same type.